src/common/storage/src/config.rs

```diff
@@ -116,6 +117,24 @@ impl StorageParams {
             StorageParams::Gcs(v) => v.endpoint_url.starts_with("https://"),
         }
     }
+
+    pub fn get_root_path(&self) -> String {
+        match self {
+            StorageParams::Azblob(v) => v.root.clone(),
+            StorageParams::Fs(v) => v.root.clone(),
+            StorageParams::Gcs(v) => v.root.clone(),
+            StorageParams::Obs(v) => v.root.clone(),
+            StorageParams::S3(v) => v.root.clone(),
+            StorageParams::Ipfs(v) => v.root.clone(),
+
+            StorageParams::Http(_) | StorageParams::Ftp(_) | StorageParams::Memory => {
+                "".to_string()
+            }
+
+            #[cfg(feature = "storage-hdfs")]
+            StorageParams::Hdfs(v) => v.root.clone(),
```

Comment:
Not a issue for this PR:

`StorageParams` becomes two different types, if `Hdfs` can be toggled on and off by a feature. A binary without feature `storage-hdfs` can not read data written by a binary with feature `storage-hdfs`.

Reply:
Tracked at https://github.com/datafuselabs/databend/issues/7936